src/Orleans/Streams/Predicates/FilterPredicateWrapperData.cs

```diff
@@ -50,6 +50,9 @@ internal class FilterPredicateWrapperData : IStreamFilterPredicateWrapper
         [NonSerialized]
         private StreamFilterPredicate predicateFunc;
 
+        // constructor used by serializator
+        private FilterPredicateWrapperData() { }
```

Comment:
@gabikliot The restriction is that to non-default constructors this solution it will pass default parameters, so checking for their correctness will most likely cause an exception. It will be only for Orleans on CoreClr, because of ```#if```. The further enhancement on the road to JUST WORK scenario would be matching constructors arguments with available parameters instead of setting those directly to the already instantiated object fields, but for now I've went with simplest solution.